<?xml version="1.0"?>
<launch>

  <arg name="map_name"              default="simple_room"/>

  <arg name="robot_name"            default="robot"/>

  <arg name="use_fake_localization" default="false"/>
  <arg name="global_planner"        default="navfn/NavfnROS"/>
  <arg name="local_planner"         default="base_local_planner/TrajectoryPlannerROS"/>
  <arg name="observation_sources"   default="laser_scan_back_observation"/>
  <arg name="xy_goal_tolerance"     default="0.1"/>
  <arg name="yaw_goal_tolerance"    default="0.05"/>

  <param name="max_experiments"     value="10"/>
  <param name="path_storage"        value=""/>

<!-- ###################################################################### -->
<!-- ###################################################################### -->
<!-- ###################################################################### -->

  <!-- simulator -->
  <include file="$(find social_worlds)/launch/gazebo.launch" >
    <arg name="model" default="$(arg map_name)" />
    <arg name="enable_gui" default="false" />
  </include>

  <!-- robot -->
  <include file="$(find hera_description)/launch/load_description.launch" />

  <!-- navigation -->
  <include file="$(find hera_bringup)/launch/bring_up.launch" >
    <arg name="map_name"              default="$(arg map_name)"/>
    <arg name="use_fake_localization" default="$(arg use_fake_localization)"/>
    <arg name="global_planner"        default="$(arg global_planner)"/>
    <arg name="local_planner"         default="$(arg local_planner)"/>
    <arg name="observation_sources"   default="$(arg observation_sources)"/>
    <arg name="xy_goal_tolerance"     default="$(arg xy_goal_tolerance)"/>
    <arg name="yaw_goal_tolerance"    default="$(arg yaw_goal_tolerance)"/>
    <arg name="enable_description"    default="false"/>
    <arg name="enable_map"            default="true"/>
    <arg name="enable_localization"   default="true"/>
    <arg name="enable_navigation"     default="true"/>
  </include>

  <!-- social_experiments -->
  <node name="social_experiments" pkg="social_experiments" type="social_experiments_node.py"
    output="screen" required="true">
    <param name="world_model_name"  value="$(arg map_name)"/>
    <param name="robot_model_name"  value="$(arg robot_name)"/>
    <param name="max_experiments"   value="$(arg max_experiments)"/>
    <param name="path_storage"      value="$(arg path_storage)"/>
    <param name="robot_vel"                value="0.3"/>
    <param name="space_factor_tolerance"   value="5"/>
    <param name="time_factor_tolerance"    value="5"/>
    <param name="path_img_freecells_start" value="$(find social_experiments)/config/map/$(arg map_name)/freecells_start.png"/>
    <param name="path_img_freecells_goal"  value="$(find social_experiments)/config/map/$(arg map_name)/freecells_goal.png"/>
  </node>

  <!-- check robot collisions -->
  <node name="hera_check_collision" pkg="hera_description" type="hera_check_collision" output="screen">
    <param name="world_name" value="default"/>
    <param name="robot_name" value="$(arg robot_name)"/>
  </node>

  <!-- real time factor publisher -->
  <node name="real_time_factor_publisher_node" pkg="social_worlds" type="real_time_factor_publisher_node" output="screen">
    <param name="world_name" value="default"/>
  </node>

  <!-- <node name="move_base_logger" pkg="rosservice" type="rosservice"
    args="call /move_base/set_logger_level {'logger: '''ros''',level: '''fatal''' '}" /> -->

</launch>
